src/main/java/net/snowflake/client/core/SFJsonResultSet.java

```diff
@@ -92,6 +92,24 @@ public Object getObject(int columnIndex) throws SFException {
     }
   }
 
+  /**
+   * Sometimes large BIGINTS overflow the java Long type. In these cases, return a BigDecimal type
+   * instead.
+   *
+   * @param columnIndex the column index
+   * @return an object of type long or BigDecimal depending on number size
+   * @throws SFException
+   */
+  private Object getBigInt(int columnIndex) throws SFException {
+    BigDecimal bigNum = getBigDecimal(columnIndex);
```

Comment:
I assume convert string into `BigDecimal` is much expensive then convert string into long. So can we find a way to only return `BigDecimal` when the number is overflow. 
This function is called by each cell so we need to make sure it is optimal in terms of performance.

Reply:
The only other ways I can think of are:
1.) Try to convert the string to a long, catch the overflow exception, and turn the string into a BigDecimal if overflow occurs. This try/catch seemed more expensive to me than the BigDecimal conversion
2.) Count the precision of the number by calculating string length. If precision >= precision of Long.MAX_VALUE, return BigDecimal else return Long. Disadvantage here is that for numbers that could be Longs but their precision= that of Long.MAX_VALUE, they will be returned as BigDecimals. This only affects a small subset of cases but it would be a behavior mismatch between Arrow and JSON. For example, 2^63-2 would be returned as a BigDecimal even though it could be returned as a Long because its precision=19, just like precision of Long.MAX_VALUE=19.

What do you think of either of these solutions?

Reply:
Those are good thoughts! I would say both try/catch and BigDecimal are expensive! I don't know which one is more expensive though. So to avoid both cases, we can use the length of the string as you said. The max/min value of LONG is +/-9,223,372,036,854,775,807 which has at least 19 digits. What we can do is if the length is smaller than 19, then directly convert to long; otherwise, do what you did here: convert to BigDecimal first, if it is outside of Long, go with BigDecimal; otherwise, getLong.

Reply:
Good idea! Done.